backend/app/services/knowledge/structured/duckdb_storage.py

```
@@ -0,0 +1,334 @@
+                f"rows={len(result_df)}, truncated={truncated}"
+            )
+
+            return {
+                "columns": list(result_df.columns),
+                "rows": result_df.values.tolist(),
+                "row_count": len(result_df),
+                "truncated": truncated,
+            }
+
+        except Exception as e:
+            logger.error(f"[DuckDB] Query failed: {e}")
```

Ruff (0.15.0)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)